larky-api/src/main/java/com/verygood/security/larkyapi/VersionedLarkyEngineImpl.java

```diff
@@ -41,29 +42,44 @@
   private static void detectVersions() {
     larkyJarByVersion = new HashMap<>();
     try {
-      PathMatchingResourcePatternResolver resolver =
-              new PathMatchingResourcePatternResolver(VersionedLarkyEngineImpl.class.getClassLoader());
-      // Get jars with format `larky-\d{>=1}.\d{>=1}.\d{>=1}-fat.jar`
-      Resource[] resources = resolver.getResources( // uses AntPathMatcher
-              "classpath*:larky-{\\d+}.{\\d+}.{\\d+}-fat.jar"
-      );
-      for (Resource resource: resources) {
-        String fileName = resource.getFilename();
-        URL fileURL = resource.getURL();
-
-        Pattern pattern = Pattern.compile("\\d+.\\d+.\\d+");
-        Matcher matcher = pattern.matcher(fileName);
-        if (matcher.find()) {
-          larkyJarByVersion.put(matcher.group(), fileURL);
-        }
+      String larky_lib = System.getProperty("user.home") + "/.larky/lib"; // default dir
+
+      String larky_alt_lib = System.getenv("LARKY_LIB_HOME");
+      if ( larky_alt_lib != null ) {
+        larky_lib = larky_alt_lib;
       }
-    } catch (Exception e) {
+
+      Stream<Path> paths = Files.walk(Paths.get(larky_lib));
+      paths
+              // Get jars with format `larky-\d{>=1}.\d{>=1}.\d{>=1}-fat.jar`
+              .filter(filePath -> filePath.toString().matches("^.*larky-\\d+.\\d+.\\d+-fat.jar$"))
+              .forEach(filePath -> {
+                try {
+                  String fileName = filePath.toString();
+                  URL fileURL = filePath.toUri().toURL();
+
+                  Pattern pattern = Pattern.compile("\\d+.\\d+.\\d+");
+                  Matcher matcher = pattern.matcher(fileName);
+                  if (matcher.find()) {
+                    larkyJarByVersion.put(matcher.group(), fileURL);
+                  }
+                } catch (Exception e) {
+                  e.printStackTrace();
+                }
+              });
+    } catch (IOException e) {
+      setupException = e;
```

Comment:
On the line below, please, consider using a logging framework instead of `printStackTrace()`. It gives you more flexibility in log configuration/routing.